pkg/operator/controllers/imageconfig/image_controller_test.go

```diff
@@ -0,0 +1,239 @@
+			arocli: arocli,
+			configcli: configfake.NewSimpleClientset(&configv1.Image{
+				ObjectMeta: imageConfigMetadata,
+				Spec: configv1.ImageSpec{
+					RegistrySources: configv1.RegistrySources{
+						BlockedRegistries: []string{
+							"quay.io", "arointsvc.azurecr.io", "arosvc.eastus.data.azurecr.io",
+						},
+					},
+				},
+			}),
+			wantConfig: `["quay.io"]`,
+		},
+		{
+			name: "Gov Cloud, function should add appropriate registries",
+			arocli: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: arov1alpha1.SingletonClusterName,
+				},
+				Spec: arov1alpha1.ClusterSpec{
+					ACRDomain:     "arointsvc.azurecr.us",
+					AZEnvironment: "AzureUSGovernmentCloud",
+					OperatorFlags: arov1alpha1.OperatorFlags{
+						ENABLED: strconv.FormatBool(true),
+					},
+					Location: "eastus",
+				},
+			}),
+			configcli: configfake.NewSimpleClientset(&configv1.Image{
+				ObjectMeta: imageConfigMetadata,
+				Spec: configv1.ImageSpec{
+					RegistrySources: configv1.RegistrySources{
+						AllowedRegistries: []string{
+							"quay.io",
+						},
+					},
+				},
+			}),
+			wantConfig: `["arointsvc.azurecr.us","arosvc.eastus.data.azurecr.us","quay.io"]`,
+		},
+		{
+			name: "AZEnvironment is Unset",
+			arocli: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: arov1alpha1.SingletonClusterName,
+				},
+				Spec: arov1alpha1.ClusterSpec{
+					OperatorFlags: arov1alpha1.OperatorFlags{
+						ENABLED: strconv.FormatBool(true),
+					},
+					Location: "eastus",
+				},
+			}),
+			configcli: configfake.NewSimpleClientset(&configv1.Image{
+				ObjectMeta: imageConfigMetadata,
+				Spec: configv1.ImageSpec{
+					RegistrySources: configv1.RegistrySources{
+						AllowedRegistries: []string{
+							"quay.io",
+						},
+					},
+				},
+			}),
+			wantConfig: `["quay.io"]`,
+		},
+		{
+			name:   "Both AllowedRegistries and BlockedRegistries are present, function should fail silently and not requeue",
+			arocli: arocli,
+			configcli: configfake.NewSimpleClientset(&configv1.Image{
+				ObjectMeta: imageConfigMetadata,
+				Spec: configv1.ImageSpec{
+					RegistrySources: configv1.RegistrySources{
+						BlockedRegistries: []string{
+							"arointsvc.azurecr.io", "arosvc.eastus.data.azurecr.io",
+						},
+						AllowedRegistries: []string{
+							"quay.io",
+						},
+					},
+				},
+			}),
+			wantErr: `both AllowedRegistries and BlockedRegistries are present`,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			r := &Reconciler{
+				log:       log,
+				arocli:    tt.arocli,
+				configcli: tt.configcli,
+			}
+			request := ctrl.Request{}
+			request.Name = "cluster"
+
+			_, err := r.Reconcile(ctx, request)
+			if err != nil && err.Error() != tt.wantErr ||
+				err == nil && tt.wantErr != "" {
+				t.Error(err)
+				t.Error("----------------------")
+				t.Error(tt.wantErr)
```

Comment:
`t.Error(err)` is sufficent: it will print test case name thanks to `t.Run(tt.name)`, line number and actual error. From that we should be able to figure out the expected error.

```suggestion
```